### PR TITLE
feat: integrate client-side validation into Select

### LINKED ISSUE #4235: [Select] Implement the new validation behavior
### An overview of the changes to be done

- Prevent the web component from setting itself to valid using `ClientValidationUtil`.
- Trigger constraint validation on client `validated` event  (requires the `HasClientValidation` interface).
- Trigger binder validation on client `validated` event (requires the `HasValidator` interface).

See [TextField](https://github.com/vaadin/flow-components/pull/3429) for an example.

### Constraints to be tested

- required

### Additional context

There are abstract classes [`AbstractValidationIT`](https://github.com/vaadin/flow-components/blob/master/vaadin-flow-components-shared-parent/vaadin-flow-components-test-util/src/main/java/com/vaadin/tests/validation/AbstractValidationIT.java) and [`AbstractValidationPage`](https://github.com/vaadin/flow-components/blob/master/vaadin-flow-components-shared-parent/vaadin-flow-components-test-util/src/main/java/com/vaadin/tests/validation/AbstractValidationPage.java) which are supposed to be extended when testing validation for a specific component.

Part of https://github.com/vaadin/platform/issues/3066

### PR DESCRIPTION
**Pull request by sissbruecker**
## Description

Makes `Select` validate on the server-side when the client attempts to validate, which means that server-side validation is now also triggered on blur. Also integrates with Binder, so that Binder validation is now also triggered when the component attempts to validate.

Closes #4235 

## Type of change

- Feature

### CODE REVIEW

File: vaadin-select-flow-parent/vaadin-select-flow-integration-tests/src/test/java/com/vaadin/flow/component/select/tests/validation/BasicValidationIT.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        $("button").id(DETACH_FIELD_BUTTON).click();
+        $("button").id(ATTACH_FIELD_BUTTON).click();
+        testField = getTestField();
+
+        assertServerInvalid();
+        assertClientInvalid();
+    }
+
+    @Test
+    public void preventWebComponentFromChangingInvalidState() {
```
Comment (vursen) [inline vaadin-select-flow-parent/vaadin-select-flow-integration-tests/src/test/java/com/vaadin/flow/component/select/tests/validation/BasicValidationIT.java:66]:
```suggestion
    public void webComponentCanNotModifyInvalidState() {
```

File: vaadin-flow-components-shared-parent/vaadin-flow-components-base/src/main/java/com/vaadin/flow/component/shared/ClientValidationUtil.java [hunk 2]
```diff
@@ -46,4 +46,21 @@ public static <C extends Component & HasValidation> void preventWebComponentFrom
 
         component.getElement().executeJs(expression.toString());
     }
+
+    public static <C extends Component & HasValidation> void preventWebComponentFromChangingInvalidState(
```
Comment (vursen) [inline vaadin-flow-components-shared-parent/vaadin-flow-components-base/src/main/java/com/vaadin/flow/component/shared/ClientValidationUtil.java:50]: **nit** Not sure if this change is worth applying :) 

```suggestion
    public static <C extends Component & HasValidation> void preventWebComponentFromModifyingInvalidState(
```

File: vaadin-select-flow-parent/vaadin-select-flow-integration-tests/src/test/java/com/vaadin/flow/component/select/tests/validation/BasicValidationIT.java (hunk 1 above)
Comment (sissbruecker) [reply vaadin-select-flow-parent/vaadin-select-flow-integration-tests/src/test/java/com/vaadin/flow/component/select/tests/validation/BasicValidationIT.java]: Done

File: vaadin-select-flow-parent/vaadin-select-flow/src/main/java/com/vaadin/flow/component/select/Select.java [hunk 3]
```diff
@@ -1045,4 +1065,45 @@ private void identifierProviderChanged(
         keyMapper.setIdentifierGetter(identifierProvider);
     }
 
+    @Override
+    protected void validate() {
+        boolean isRequired = this.isRequiredIndicatorVisible();
```
Comment (vursen) [inline vaadin-select-flow-parent/vaadin-select-flow/src/main/java/com/vaadin/flow/component/select/Select.java:1070]: **nit:** Could use `ValidationUtil.checkRequired` for the sake of consistency.

```suggestion
        ValidationResult requiredValidation = ValidationUtil.checkRequired(isRequiredIndicatorVisible(), getValue(), getEmptyValue());
        setInvalid(requiredValidation.isError());
```

Comment (sissbruecker) [reply vaadin-select-flow-parent/vaadin-select-flow/src/main/java/com/vaadin/flow/component/select/Select.java:1071]: Done

File: vaadin-select-flow-parent/vaadin-select-flow/src/main/java/com/vaadin/flow/component/select/Select.java [hunk 4]
```diff
@@ -126,6 +134,12 @@ public Select() {
         setPresentationValue(null);
 
         getElement().appendChild(listBox.getElement());
+
+        addValueChangeListener(e -> validate());
```
[the remark this answers is not in the document]
Comment (sissbruecker) [reply vaadin-select-flow-parent/vaadin-select-flow/src/main/java/com/vaadin/flow/component/select/Select.java:136]: Removed the feature flag checks as discussed.

**Review by vursen — APPROVED** [review]

**Pull request merged**